### PR TITLE
[enterprise-4.15] TELCODOCS-1838 Update reference SiteConfig to use rootDeviceHints wit…

### PR DESCRIPTION
**Pull request by kquinn1204**
[TELCODOCS-1838]: Docs: Create partitions through SiteConfig

Version(s):
4.15

Issue:https://issues.redhat.com/browse/TELCODOCS-1832

Link to docs preview:
https://79810--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/ztp-advanced-policy-config.html
https://79810--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/ztp-deploying-far-edge-sites.html
https://79810--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/ztp-precaching-tool.html

QE review:
- [x] QE has approved this change.

Additional information:
“Cherry Picked from d59e7cf02bd625fb56cd46e16d91b48c6859a07b xref: [https://github.com/openshift/openshift-docs/pull/75726]

### CODE REVIEW

**Comment by kquinn1204** [comment]
@sabbir-47 can you PTAL at this cherrypick to 4.15

File: snippets/ztp_example-sno.yaml [hunk 1]
```diff
@@ -12,96 +12,140 @@ spec:
...
+        # disk partition at `/var/lib/containers` with ignitionConfigOverride. Some values must be updated. See DiskPartitionContainer.md for more details
+        ignitionConfigOverride: | 
+           {
+            "ignition": {
+              "version": "3.2.0"
+            },
+            "storage": {
+              "disks": [
+                {
+                  "device": "/dev/disk/by-id/wwn-0x6b07b250ebb9d0002a33509f24af1f62",
```
Comment (sabbir-47) [inline snippets/ztp_example-sno.yaml:87]: please change the devices by-id to by-path like below
"device": "/dev/disk/by-path/pci-0000:01:00.0-scsi-0:2:0:0"

**Comment by sabbir-47** [comment]
Added a comment, otherwise it lgtm.

File: modules/ztp-configuring-disk-partitioning.adoc [hunk 2]
```diff
@@ -8,39 +8,197 @@
...
 
 .Prerequisites
 
-* You have installed the OpenShift CLI (`oc`).
+* Install Butane.
+
+.Procedure
 
-* You have logged in to the hub cluster as a user with `cluster-admin` privileges.
+. Create the `storage.bu` file.
```
Comment (lpettyjo) [inline modules/ztp-configuring-disk-partitioning.adoc:22]: Recommend adding "using the following example YAML file:"

**Review by lpettyjo — COMMENTED** [review]
You have two commits. Please squash to one.

One other comment; Otherwise, LGTM!

**Review by sheriff-rh — APPROVED** [review]
Changes look good; merging.

**Pull request merged**